src/yb/tools/yb-admin-test.cc

```diff
@@ -682,6 +682,130 @@ TEST_F(AdminCliTest, TestModifyTablePlacementPolicy) {
     extra_table, ClusterVerifier::EXACTLY, rows_inserted));
 }
 
+TEST_F(AdminCliTest, TestModifyPlacementInfoWithDifferentReplicas) {
+  // Start a cluster with 3 tservers, each corresponding to a different zone.
+  FLAGS_num_tablet_servers = 3;
+  FLAGS_num_replicas = 2;
+  std::vector<std::string> master_flags;
+  master_flags.push_back("--enable_load_balancing=true");
+  master_flags.push_back("--catalog_manager_wait_for_new_tablets_to_elect_leader=false");
+  std::vector<std::string> ts_flags;
+  ts_flags.push_back("--placement_cloud=c");
+  ts_flags.push_back("--placement_region=r");
+  ts_flags.push_back("--placement_zone=z${index}");
+  BuildAndStart(ts_flags, master_flags);
+
+  const std::string& master_address = ToString(cluster_->master()->bound_rpc_addr());
+  auto client = ASSERT_RESULT(YBClientBuilder()
+      .add_master_server_addr(master_address)
+      .Build());
+
+  // Modify the cluster placement policy to consist of 2 zones of 2 and 3
+  // replicas respectively.
+  std::string output;
+  ASSERT_OK(Subprocess::Call(ToStringVector(GetAdminToolPath(),
+   "-master_addresses", master_address, "modify_placement_info", "c.r.z0:2,c.r.z1:3", 2, ""),
```

Comment:
In general, the RF (2) should equal the sum of all the num replicas per placement (2 + 3). I think the easiest way to test this would be to put 2 ts in c.r.z0 and 2 ts in c.r.z1 (see ts_flags), define a placement like c.r.z0:1,c.r.z1:2, and that for each tablet, there are 2 copies in z1 and 1 in z0.